### PR TITLE
gh-92771: Add pathlib.Path.rmtree method

### PR DESCRIPTION
**Pull request by zmievsa**
https://github.com/python/cpython/issues/92771

* Issue: gh-92771

### CODE REVIEW

File: Doc/library/pathlib.rst [hunk 1]
```diff
@@ -1242,6 +1242,16 @@ call fails (for example because the path doesn't exist).
...
+.. method:: Path.rmtree()
+
+   Recursively delete a directory tree.
+
+   if *on_error* is not ``None``, it is called to handle the error with arguments
+   (func, path, exc_info) where func is platform and implementation
+   dependent; path is the argument to that function that caused it to
+   fail; and exc_info is a tuple returned by :func:`sys.exc_info`.
+
+   Unlike :func:`shutil.rmtree`, *on_error* to ``None`` means ignore all errors.
```
Comment (merwok) [inline Doc/library/pathlib.rst:1254]: Add versionadded note

File: Doc/library/pathlib.rst [hunk 2]
```diff
@@ -1295,7 +1305,6 @@ call fails (for example because the path doesn't exist).
 
    .. versionadded:: 3.10
 
-
```
Comment (merwok) [inline Doc/library/pathlib.rst:1298]: Please don’t make unrelated changes

**Comment by brettcannon** [comment]
Closing as per https://github.com/python/cpython/issues/92771#issuecomment-1293806841 . Thanks @Ovsyanka83 for giving this a try!
